astropy/table/table.py

```diff
@@ -3274,6 +3274,16 @@ def insert_row(self, index, vals=None, mask=None):
         else:
             raise TypeError("Vals must be an iterable or mapping or None")
 
+        if N == 0 and any(
```

Comment:
Shouldn't this apply for a non-zero length table as well? That would have been the case for putting the warning into `Column.insert`.

Reply:
I'm following @mhvk's suggestion from https://github.com/astropy/astropy/pull/16004#issuecomment-1942915020
In short, the idea is that we make this as small-scoped as possible because it's not an ideal solution, so there's a high chance we'll revisit later.

Reply:
OK, fair enough, stick with N=0.